cpusupport/cpusupport.h

```diff
@@ -126,7 +126,7 @@ CPUSUPPORT_FEATURE(x86, aesni, X86_AESNI);
 CPUSUPPORT_FEATURE(x86, rdrand, X86_RDRAND);
 CPUSUPPORT_FEATURE(x86, shani, X86_SHANI);
 CPUSUPPORT_FEATURE(x86, sse2, X86_SSE2);
-CPUSUPPORT_FEATURE(x86, sse42, X86_SSE42_64);
+CPUSUPPORT_FEATURE(x86, sse42, X86_SSE42);
```

Comment:
I'm confused, you're replacing X86_SSE42_64 with X86_SSE42 in this file, but in `cpusupport.sh` you still have both?

Reply:
That is correct.  It's a subtle point that I didn't sufficiently clarify in the commit messages (and/or direct comments in the code).

`cpusupport_x86_sse42.c` checks bit 20 of ECX.  This tells us if the CPU supports SSE4.2.  It doesn't say anything about whether it's in 64-bit mode.

Further investigation reveals that we could put 0x80000001 in EAX and check bit 29 of EDX, which tells us if the CPU is in "Long mode".  So if we wanted to, we could have a separate `cpusupport_x86_sse42_64.c` which checks both bits to ensure that we're in 64-bit mode and have SSE4.2 available.

However, given the current design of `crc32_sse42.c`, I don't see the need for a runtime detection of `sse42` vs. `sse42_64`.  As long as the CPU has sse4.2, we can call `CRC32C_Update_SSE42()`.  If the environment supports 64-bit code at compile-time, that function will include 64-bit instructions; if it doesn't, then it'll only use 32-bit instructions.

Granted, there's cross-compiling to consider.  But `_mm_crc32_u64()` is hidden within `__x86_64__` inside `<smmintr.h>`, and if we compile with `-m32` then that's not enabled.

Reply:
Got it.  Right, one run-time test for SSE4.2 since it's a single CPU feature, but two compile-time tests since they use different intrinsics.